src/Pos/Explorer/BListener.hs

```diff
@@ -0,0 +1,247 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Blockchain listener for Explorer.
+-- Callbacks on application and rollback.
+
+module Pos.Explorer.BListener
+       ( runExplorerBListener -- (+) BListener instance
+       ) where
+
+import           Control.Monad.Trans.Identity (IdentityT (..))
+import           Data.Coerce                  (coerce)
+import qualified Ether
+
+import           Pos.Block.Types              (Blund)
+import           Pos.DB.BatchOp               (SomeBatchOp)
+import           Pos.Ssc.Class.Helpers        (SscHelpersClass)
+import           Pos.Util.Chrono              (NE, NewestFirst (..),
+                                               OldestFirst (..))
+import           Universum
```

Comment:
Well, the link is in [README](https://github.com/input-output-hk/cardano-sl#for-contributors), also it can be found in [cardano-sl-style-guides repository](https://github.com/input-output-hk/cardano-sl-style-guides/blob/master/haskell-style-guide.md).